src/main/java/com/cse/helium/app/services/keyvault/KeyVaultService.java

```diff
@@ -119,24 +119,33 @@ public KeyVaultService(IEnvironmentReader environmentReader)
     .buildAsyncClient();
   }
 
-  private HttpLogDetailLevel getKvLogDetailLevel() {
+  private KeyVaultSecretsLogPolicy getKvLogPolicy() {
     
     Level currentLogLevel = logger.getLevel();
-    HttpLogDetailLevel returnLevel = HttpLogDetailLevel.NONE;
+    HttpLogDetailLevel logLevel = HttpLogDetailLevel.NONE;
 
     if (currentLogLevel == Level.TRACE 
-        || currentLogLevel == Level.DEBUG 
-        || currentLogLevel == Level.INFO) {
+        || currentLogLevel == Level.DEBUG) {
+
+      logLevel = HttpLogDetailLevel.BODY_AND_HEADERS;
+
+
+    } else if (currentLogLevel == Level.INFO) {
+
+      logLevel = HttpLogDetailLevel.HEADERS;
 
-      returnLevel = HttpLogDetailLevel.HEADERS;
     } else if (currentLogLevel == Level.WARN 
           || currentLogLevel == Level.ERROR
           || currentLogLevel == Level.FATAL) {
 
-      returnLevel = HttpLogDetailLevel.BASIC;
+      logLevel = HttpLogDetailLevel.BASIC;
```

Comment:
BASIC by definitionsLogs everything in BASIC, plus all the request and response headers. Should it apply to WARN as well?

Reply:
It does apply to WARN based on the code about.  Also, it doesn't log as you note, I placed a matrix of what is logged in #230 .